### PR TITLE
Fixtures

### PR DESCRIPTION
**Pull request by Yurasb**
1. adding fixtures for purge, create and redirect
2. optimizing preconditions
3. added tests for empty links list

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import pytest
+import requests
+
+
+@pytest.fixture()
+def purge_all_links(request):
+    requests.request(
```
Comment (v1-wizard) [inline tests/conftest.py]: requests.delete(..) - looks more cool.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        method='DELETE',
+        url='http://localhost:8888/admin/all_links',
+        data='{"Are you sure?":"Yes"}'
+    )
+
+
+@pytest.fixture()
+def create_shortcut_link(request):
+    create = requests.request(
```
Comment (v1-wizard) [inline tests/conftest.py]: requests.post(...)

File: tests/test_delete_all_links.py [hunk 3]
```diff
@@ -2,13 +2,9 @@
...
 
-def test_delete_all_links_status_code():
+def test_delete_all_links_status_code(create_shortcut_link):
     # precondition
-    request(
-        method='POST',
-        url='http://localhost:8888/shortcut',
-        data='{ "link": "https://github.com/Yurasb/url_shortener_testing"}'
-    )
+    create_shortcut_link
```
Comment (v1-wizard) [inline tests/test_delete_all_links.py]: Wtf? I think you could delete this "#precondition" section.

File: tests/test_delete_all_links.py [hunk 4]
```diff
@@ -2,13 +2,9 @@
 from requests import request
```
Comment (v1-wizard) [inline tests/test_delete_all_links.py]: Could you add test that really check that all links is delete from db?

File: tests/test_list_all_links.py [hunk 5]
```diff
@@ -78,7 +91,9 @@ def test_all_links_wrong_method_body():
     v = Validator(
         {
             'status': {'type': 'integer', 'allowed': [405]},
-            'message': {'type': 'string', 'allowed': ['Method Not Allowed']}
+            'message': {
+                'type': 'string', 'allowed': ['Method Not Allowed']
+            }
         }
     )
     assert v.validate(response.json())
```
Comment (v1-wizard) [inline tests/test_list_all_links.py]: I think you should make this assert more verbose. 
`assert v.validate(response.json()), v.errors`

File: tests/test_redirect.py [hunk 6]
```diff
@@ -4,107 +4,77 @@
...
-    )
-    create = request(
-        method='POST',
-        url='http://localhost:8888/shortcut',
-        data='{ "link": "https://github.com/Yurasb/url_shortener_testing"}'
-    )
+    purge_all_links
+    create_shortcut_link
     # action
     response = request(
```
Comment (v1-wizard) [inline tests/test_redirect.py]: You should import this things as:
`import requests`
or
`from requests import request`
and use the same in all code. It's called code style. =)

File: data/schema.xsd [hunk 7]
```diff
@@ -0,0 +1,968 @@
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
```
Comment (v1-wizard) [inline data/schema.xsd]: Is this full scheme for check html as xml? I think it's very powerful solution little url_shortener. Maybe use something lite? For example: `https://docs.python.org/3/library/html.parser.html`

**Review by v1-wizard — CHANGES_REQUESTED** [review]
If you have some question you could contact with me in slack.

File: tests/test_delete_all_links.py (hunk 3 above)
Comment (Yurasb) [reply tests/test_delete_all_links.py]: @v1-wizard Why? We create some shortcut to be sure that we really delete something with delete request

File: data/schema.xsd (hunk 7 above)
Comment (Yurasb) [reply data/schema.xsd]: @v1-wizard I would prefer to use lxml, but assert not by XSD - which is really overhead - but using XPath for particular element on the page to ensure that we get expected page.

File: tests/test_delete_all_links.py (hunk 4 above)
Comment (Yurasb) [reply tests/test_delete_all_links.py]: @v1-wizard How about try to redirect by deleted id? I think it will show that shortcut is really deleted

**Comment by Yurasb** [comment]
Closing to recreate with all changes

**Pull request closed without merging**